### PR TITLE
Introducing the Gtester Plugin [V2]

### PR DESCRIPTION
**Pull request by apahim**
v2:
- Make the tests list unique.

v1: #2457

### CODE REVIEW

File: optional_plugins/gtester/avocado_gtester/__init__.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+class GtesterLoader(loader.TestLoader):
+    """
+    Gtester loader class
+    """
+    name = "gtester"
+
+    def __init__(self, args, extra_params):
+        super(GtesterLoader, self).__init__(args, extra_params)
+
+    def discover(self, url, which_tests=loader.DEFAULT):
```
Comment (clebergnu) [inline optional_plugins/gtester/avocado_gtester/__init__.py]: A nitpick: the base class had the `url` argument renamed to `reference`.  It makes sense to change it here too (and maybe on all other loader classes).

File: optional_plugins/gtester/avocado_gtester/__init__.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            result = process.run(cmd)
+        except Exception as details:
+            if which_tests == loader.ALL:
+                return [(NotGtesterTest, {"name": "%s: %s" % (url, details)})]
+            return []
+
+        for test in result.stdout.splitlines():
+            test_name = "%s:%s" % (url, test)
+            try:
+                # Avoiding duplicates
```
Comment (clebergnu) [inline optional_plugins/gtester/avocado_gtester/__init__.py]: Can you please create a card to attempt to fix the issue upstream and reference it here?

File: optional_plugins/gtester/avocado_gtester/__init__.py (hunk 1 above)
Comment (apahim) [reply optional_plugins/gtester/avocado_gtester/__init__.py]: sure

File: optional_plugins/gtester/avocado_gtester/__init__.py (hunk 2 above)
Comment (apahim) [reply optional_plugins/gtester/avocado_gtester/__init__.py]: ok

**Pull request closed without merging**